tidy3d/components/tcad/materials/charge.py

```diff
@@ -0,0 +1,97 @@
+from typing import Union
+
+import pydantic.v1 as pd
+
+from tidy3d.components.data.data_array import SpatialDataArray
+from tidy3d.components.tcad.bandgap import SlotboomNarrowingBandGap
+from tidy3d.components.tcad.generation_recombination import (
+    AugerRecombination,
+    RadiativeRecombination,
+    ShockleyReedHallRecombination,
+)
+from tidy3d.components.tcad.materials.abstract import AbstractHeatChargeSpec
+from tidy3d.components.tcad.mobility import CaugheyThomasMobility
+from tidy3d.components.tcad.types import (
+    BandGapModelTypes,
+    MobilityModelTypes,
+    RecombinationModelTypes,
+)
+from tidy3d.constants import ELECTRON_VOLT
+
+
+class ElectronicSpec(AbstractHeatChargeSpec):
+    # TODO does this account for the whole temporal dynamics? If not then it should be called ElectroStatic, which it was called before, but unsure why it got changed?
+    """
+    This class is used to define electro-static semiconductors.
+
+    Notes
+    -----
+        Both acceptors and donors can be either a positive number or an 'xarray.DataArray'.
+        Default values for parameters and models are those appropriate for Silicon
+    """
+
+    electrons_effective_density: pd.PositiveFloat = pd.Field(
+        2.86e19,
+        title="Effective density of electron states",
+        description="Effective density of electron states",
+        units="cm^(-3)",
+    )
+    N_e = electrons_effective_density
+
+    holes_effective_density: pd.PositiveFloat = pd.Field(
```

Comment:
These names are confusing. The literature universally calls this effective density of states with Nc and Nv as symbols.

Reply:
Broadly inclined to bring back your standard naming on this and maybe just make the description longer explaining it more.